apps/smart-contracts/core/test/unit/PrePOMarket.test.ts

```diff
@@ -460,122 +462,286 @@ describe('=> prePOMarket', () => {
+      const expectedPartialFeeAmount = expectedMintingFeeAmount.div(2)
+      expect(expectedPartialFeeAmount).gt(0)
+
+      await prePOMarket.connect(user).mint(TEST_MINT_AMOUNT, recipient.address)
 
+      expect(await collateralToken.balanceOf(user.address)).eq(
+        userCollateralBefore.sub(TEST_MINT_AMOUNT)
+      )
+      expect(await collateralToken.balanceOf(testTreasury)).eq(
+        treasuryCollateralBefore.add(expectedPartialFeeAmount)
+      )
+      expect(await longToken.balanceOf(recipient.address)).eq(
+        recipientLongTokenBalanceBefore.add(TEST_MINT_AMOUNT.sub(expectedPartialFeeAmount))
+      )
+      expect(await shortToken.balanceOf(recipient.address)).eq(
+        recipientShortTokenBalanceBefore.add(TEST_MINT_AMOUNT.sub(expectedPartialFeeAmount))
+      )
+    })
+
+    it('mints using amount with full fee subtracted if hook takes full fee', async () => {
+      const userCollateralBefore = await collateralToken.balanceOf(user.address)
+      const treasuryCollateralBefore = await collateralToken.balanceOf(treasury.address)
+      const recipientLongTokenBalanceBefore = await longToken.balanceOf(recipient.address)
+      const recipientShortTokenBalanceBefore = await shortToken.balanceOf(recipient.address)
+      expect(expectedMintingFeeAmount).gt(0)
+
+      await prePOMarket.connect(user).mint(TEST_MINT_AMOUNT, recipient.address)
+
+      expect(await collateralToken.balanceOf(user.address)).eq(
+        userCollateralBefore.sub(TEST_MINT_AMOUNT)
+      )
+      expect(await collateralToken.balanceOf(treasury.address)).eq(
+        treasuryCollateralBefore.add(expectedMintingFeeAmount)
+      )
+      expect(await longToken.balanceOf(recipient.address)).eq(
+        recipientLongTokenBalanceBefore.add(expectedLongShortAmount)
+      )
+      expect(await shortToken.balanceOf(recipient.address)).eq(
+        recipientShortTokenBalanceBefore.add(expectedLongShortAmount)
+      )
+    })
+
+    it('calls hook with correct parameters if funder = recipient', async () => {
       await prePOMarket.connect(user).mint(TEST_MINT_AMOUNT, user.address)
 
       expect(mintHook.hook).calledWith(
         user.address,
         user.address,
         TEST_MINT_AMOUNT,
-        TEST_MINT_AMOUNT
+        expectedLongShortAmount
       )
     })
 
     it('calls hook with correct parameters if funder != recipient', async () => {
       expect(user.address).not.eq(recipient.address)
-      await collateralToken.connect(deployer).transfer(user.address, TEST_MINT_AMOUNT)
-      await collateralToken.connect(user).approve(prePOMarket.address, TEST_MINT_AMOUNT)
 
       await prePOMarket.connect(user).mint(TEST_MINT_AMOUNT, recipient.address)
 
       expect(mintHook.hook).calledWith(
         user.address,
         recipient.address,
         TEST_MINT_AMOUNT,
-        TEST_MINT_AMOUNT
+        expectedLongShortAmount
       )
     })
 
     it('ignores hook if not set', async () => {
```

Comment:
```suggestion
    it('ignores hook if not set and fee percent > 0%', async () => {
```
(and add check that fee percent > 0%)

so that we cover both the =0% and >0% cases